scripts/inference/endpoint_generate.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+import asyncio
+import copy
+import logging
+import math
+import os
+import time
+from argparse import ArgumentParser, Namespace
+from typing import cast
+
+import pandas as pd
+import requests
+from composer.utils import (ObjectStore, maybe_create_object_store_from_uri,
+                            parse_uri)
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")
+log = logging.getLogger(__name__)
+
+ENDPOINT_API_KEY_ENV: str = 'ENDPOINT_API_KEY'
+ENDPOINT_URL_ENV: str = 'ENDPOINT_URL'
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    parser = ArgumentParser(
+        description='Call prompts against a text completions endpoint')
+
+    #####
+    # Path Parameters
+    parser.add_argument(
+        '-p',
+        '--prompts',
+        nargs='+',
+        help='Generation prompts. Use syntax "file::/path/to/prompt.txt" to load a ' +\
+             'prompt contained in a txt file.'
+        )
+
+    now = time.strftime("%Y%m%d-%H%M%S")
+    default_local_folder = f"/tmp/output/{now}"
+    parser.add_argument('-l',
+                        '--local-folder',
+                        type=str,
+                        default=default_local_folder,
+                        help="Local folder to save the output")
+
+    parser.add_argument('-o',
+                        '--output-folder',
+                        help="Remote folder to save the output")
+
+    #####
+    # Generation Parameters
+    parser.add_argument(
+        '--rate-limit',
+        type=int,
+        default=5,
+        help="Max number of calls to make to the endpoint in a second")
+    parser.add_argument(
+        '--batch-size',
+        type=int,
+        default=5,
+        help="Max number of calls to make to the endpoint in a single request")
+
+    #####
+    # Endpoint Parameters
+    parser.add_argument(
+        '-e',
+        '--endpoint',
+        type=str,
+        help=
+        f'OpenAI-compatible text completions endpoint to query on. If not set, will read from {ENDPOINT_URL_ENV}'
+    )
+
+    parser.add_argument('--max-tokens', type=int, default=100)
+    parser.add_argument('--temperature', type=float, default=1.0)
+    parser.add_argument('--top-k', type=int, default=50)
+    parser.add_argument('--top-p', type=float, default=1.0)
+    return parser.parse_args()
+
+
+def load_prompt_string_from_file(prompt_path_str: str):
```

Comment:
I realize that this is being re-used from `hf_generate`, so we don't want them to diverge, but it would be so useful if we could allow for files that contain multiple prompts. It could be as simple as just defining some sort of split delimiter string, using it to do `prompt_string.split(PROMPT_DELIMITER)`, and modifying the script to expect this function to return a list of strings instead of a single string.